tests/publiccloud/ipa.pm

```diff
@@ -15,13 +15,33 @@ use base "opensusebasetest";
 use strict;
 use testapi;
 use utils;
-use File::Basename;
-use registration 'add_suseconnect_product';
 use serial_terminal 'select_virtio_console';
-use version_utils qw(is_sle is_opensuse);
 
+sub get_ipa_tests {
+    my $ipa_tests = get_required_var('PUBLIC_CLOUD_IPA_TESTS');
+    $ipa_tests =~ s/,/ /g;
+    return $ipa_tests;
+}
+
+sub save_logs {
+    my $ret = script_run("test -d ipa_results");
+    return if (!defined($ret) || $ret != 0);
+
+    my $image = get_required_var('PUBLIC_CLOUD_IMAGE_ID');
+
+    my $output = script_output("find ipa_results -type f");
```

Comment:
NFV perf scenarios facing same task which is "collect logs from 3d party tool running inside openQA and upload them with upload_logs" . Take a look how it done 
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/nfv/run_performance_tests.pm#L41
 for your implementation simpliest reason to not do it like this is that you should keep in mind that behind each `upload_logs` function call stands action when backend typing really long url , calling curl and then waiting for upload end. 
But maybe you don't have other choice because of reasons which I miss to understand , so let's discuss this tomorrow

Reply:
Ok.  Thx for the hint, I will ping you tomorrow.
Just one comment, at least the *.results file need to be uploaded with parse_extra_log().